### DISCUSSION TITLE
Failing to get started with V3 - vite error

### DISCUSSION THREAD

**Discussion by quirm** [sanity-io/sanity#3679] [Sanity Studio Support] [answered]
Hello, I would like to use v3 as it seems to be the future but I'm not even able to start an empty project as described in docs.

I tried this:
```
npm create sanity@dev-preview
```

Then, in another attempt, I removed the globally installed CLI and installed `@sanity/cli@3.0.0-dev-preview.17` and did `sanity init`. Any configuration from CLI produces a project that after `yarn start` ends with a vite related error:
```
[plugin:vite:esbuild] Transform failed with 1 error:
/Users/user/cms-sanity/empty-test/schemas/index.ts:1:7: ERROR: Transforming const to the configured target environment ("es5") is not supported yet
/Users/user/cms-sanity/empty-test/schemas/index.ts:1:7
Transforming const to the configured target environment ("es5") is not supported yet
1  |  export const schemaTypes = []
   |         ^
2  |
```

There is no customization, everything is installed out of the box as described with the clean slate. Could anyone give me a hint what could be a problem?

**Comment by quirm** [comment]
In case someone faces the same issue. `tsconfig.json` is generated by sanity cli with `target: "es5"`, changing it to es6 or esnext will fix the issue with vite.

**Reply by stevenbister** [reply]
Hope you don't mind me expanding a little further; I found that using `es6` will cause vite to throw a warning when trying to use `import.meta`

```
[vite] warning: "import.meta" is not available in the configured target environment ("es6") and will be empty
7  |    title: 'OSC',
8  |
9  |    projectId: import.meta.env.SANITY_STUDIO_API_PROJECT_ID,
   |               ^
10 |    dataset: import.meta.env.SANITY_STUDIO_API_DATASET
```

using `target: "esnext"` solves that issue

**Comment by rexxars** [comment, marked as answer]
Should be fixed for new projects in [v3.0.0-dev-preview.19](https://github.com/sanity-io/sanity/releases/tag/v3.0.0-dev-preview.19) - for existing projects, set the `compilerOptions.target` to `es2017` in your `tsconfig.json`
